test/src/main/java/org/apache/accumulo/test/functional/BulkLoadIT.java

```diff
@@ -127,27 +130,42 @@ private void testSingleTabletSingleFile(AccumuloClient c, boolean offline) throw
 
     String h1 = writeData(dir + "/f1.", aconf, 0, 332);
 
-    c.tableOperations().importDirectory(dir).to(tableName).load();
+    c.tableOperations().importDirectory(dir).to(tableName).tableTime(setTime).load();
 
     if (offline)
       c.tableOperations().online(tableName);
 
-    verifyData(c, tableName, 0, 332);
+    verifyData(c, tableName, 0, 332, setTime);
     verifyMetadata(c, tableName,
         ImmutableMap.of("0333", ImmutableSet.of(h1), "null", ImmutableSet.of()));
   }
 
   @Test
   public void testSingleTabletSingleFile() throws Exception {
     try (AccumuloClient client = createAccumuloClient()) {
-      testSingleTabletSingleFile(client, false);
+      testSingleTabletSingleFile(client, false, false);
+    }
+  }
+
+  @Test
+  public void testSetTime() throws Exception {
+    try (AccumuloClient client = createAccumuloClient()) {
+      tableName = "testSetTime_table1";
+      NewTableConfiguration newTableConf = new NewTableConfiguration();
+      // set logical time type so we can set time on bulk import
+      newTableConf.setTimeType(TimeType.LOGICAL);
+      client.tableOperations().create(tableName, newTableConf);
+      aconf = getCluster().getServerContext().getConfiguration();
+      fs = getCluster().getFileSystem();
+      rootPath = cluster.getTemporaryPath().toString();
+      testSingleTabletSingleFile(client, false, true);
     }
   }
 
   @Test
   public void testSingleTabletSingleFileOffline() throws Exception {
     try (AccumuloClient client = createAccumuloClient()) {
-      testSingleTabletSingleFile(client, true);
+      testSingleTabletSingleFile(client, true, false);
```

Comment:
See, this is where boolean params start really being terrible... when you have multiple in a row, and it's impossible to tell what's intended by looking at the caller alone. I know this is "just a test", but an EnumSet of option flags would be more appropriate here than multiple booleans.

Reply:
I don't think its terrible with one or two.  Any more then for sure.  It also helps to have an IDE that prepens the params with their name. :)

I can open a follow on might be a good help wanted ticket.

Reply:
Probably not worth a follow-up in this case. I only mention it to contrast a "good" use of boolean in the fluent API, vs. (what I see as) a "bad" use here in this lots-of-flags-as-separate-params method.

Reply:
Ah OK gotcha